File: solutions/551.student-attendance-record-i/student-attendance-record-i.js
/**
 * @param {string} s
 * @return {boolean}
 */
var checkRecord = function(s) {
    let absentCnt = 0;
    for(let i = 0; i < s.length; i++){
        if(s[i] === 'A'){
            absentCnt++;
            if(absentCnt > 1){
                return false;
            }
        }
        if(i > 1 && s[i] === 'L' && s[i - 1] === 'L' && s[i - 2] === 'L'){
            return false;
        }
    }
    return true;
};

File: solutions/551.student-attendance-record-i/student-attendance-record-i.java
class Solution {
    public boolean checkRecord(String s) {
        char[] chs = s.toCharArray();
        int absentCnt = 0;
        for(int i = 0; i < s.length(); i++){
            if(chs[i] == 'A'){
                absentCnt++;
                if(absentCnt > 1){
                    return false;
                }
            }
            if(i > 1 && chs[i] == 'L' && chs[i - 1] == 'L' && chs[i - 2] == 'L'){
                return false;
            }
        }
        return true;
    }
}